src/main/java/com/conveyal/analysis/datasource/GeoTiffDataSourceIngester.java

```diff
@@ -0,0 +1,92 @@
+package com.conveyal.analysis.datasource;
+
+import com.conveyal.analysis.models.Bounds;
+import com.conveyal.analysis.models.DataSource;
+import com.conveyal.analysis.models.SpatialDataSource;
+import com.conveyal.r5.analyst.progress.ProgressListener;
+import org.geotools.coverage.GridSampleDimension;
+import org.geotools.coverage.grid.GridCoverage2D;
+import org.geotools.coverage.grid.GridEnvelope2D;
+import org.geotools.coverage.grid.io.AbstractGridFormat;
+import org.geotools.coverage.grid.io.GridFormatFinder;
+import org.geotools.geometry.jts.ReferencedEnvelope;
+import org.geotools.referencing.CRS;
+import org.geotools.referencing.crs.DefaultGeographicCRS;
+import org.geotools.util.factory.Hints;
+import org.opengis.coverage.SampleDimensionType;
+import org.opengis.referencing.FactoryException;
+import org.opengis.referencing.crs.CoordinateReferenceSystem;
+import org.opengis.referencing.operation.MathTransform;
+import org.opengis.referencing.operation.TransformException;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import static com.conveyal.file.FileStorageFormat.GEOTIFF;
+import static com.conveyal.r5.util.ShapefileReader.GeometryType.PIXEL;
+
+/**
+ * GoeTIFFs are used as inputs in network building as digital elevation profiles, and eventually expected to
+ * serve as impedance or cost fields (e.g. shade bonus and pollution malus).
+ */
+public class GeoTiffDataSourceIngester extends DataSourceIngester {
+
+    private final SpatialDataSource dataSource;
+
+    public GeoTiffDataSourceIngester () {
+        this.dataSource = new SpatialDataSource();
+        dataSource.geometryType = PIXEL;
+        dataSource.fileFormat = GEOTIFF; // Should be GEOTIFF specifically
+    }
+
+    @Override
+    protected DataSource dataSource () {
+        return dataSource;
+    }
+
+    @Override
+    public void ingest (File file, ProgressListener progressListener) {
+        progressListener.beginTask("Processing uploaded GeoTIFF", 1);
+        AbstractGridFormat format = GridFormatFinder.findFormat(file);
+        Hints hints = new Hints(Hints.FORCE_LONGITUDE_FIRST_AXIS_ORDER, Boolean.TRUE);
+        var coverageReader = format.getReader(file, hints);
+        GridCoverage2D coverage;
+        try {
+             coverage = coverageReader.read(null);
+        } catch (IOException e) {
+            throw new DataSourceException("Could not read GeoTiff.", e);
+        }
+        // Transform to WGS84 to ensure this will not trigger any errors downstream.
+        CoordinateReferenceSystem coverageCrs = coverage.getCoordinateReferenceSystem2D();
+        MathTransform wgsToCoverage, coverageToWgs;
+        ReferencedEnvelope wgsEnvelope;
+        try {
+            wgsToCoverage = CRS.findMathTransform(DefaultGeographicCRS.WGS84, coverageCrs);
+            coverageToWgs = wgsToCoverage.inverse();
```

Comment:
`coverageToWgs` appears to be unused.

Reply:
I wanted to validate that both transforms can be created to detect any problems early. Transforms in both directions will be used - for example, below the wgsEnvelope is performing this transform. So I guess this call is redundant since creating the wgsEnvelope would fail if GeoTools doesn't know how to convert the CRS to WGS84.

Reply:
I added a comment to explain why these unused transforms are looked up.